data_access_layer/client.py

```diff
@@ -0,0 +1,69 @@
+from typing import Any, cast
+
+from gql import Client, gql
+from gql.transport.aiohttp import AIOHTTPTransport
+
+from settings import DAL_GRAPHQL_AUTH_KEY, DAL_GRAPHQL_ENDPOINT
+
+
+class _CanvasGQLClient:
+    """
+    This is a GraphQL client that can be used to query home-app in order to fetch data for use in plugins.
+
+    Usage Examples:
+
+    A query with no parameters:
+
+    TEST_QUERY_NO_PARAMS = '''
+      query PatientsAll {
+        patients {
+          edges {
+            node {
+              firstName
+              lastName
+              birthDate
+            }
+          }
+        }
+      }
+    '''
+
+    client = _CanvasGQLClient()
+    result = client.query(TEST_QUERY_NO_PARAMS)
+    print(result) # returns dictionary
+
+    A query with parameters:
+
+    TEST_QUERY_WITH_PARAMS = '''
+        query PatientGet($patientKey: String!) {
+          patient(patientKey: $patientKey) {
+            firstName
+            lastName
+            birthDate
+          }
+        }
+    '''
+
+    client = _CanvasGQLClient()
+    result = client.query(TEST_QUERY_NO_PARAMS)
+    print(result)
+    """
+
+    def __init__(self) -> None:
+        transport = AIOHTTPTransport(
+            url=cast(str, DAL_GRAPHQL_ENDPOINT),
+            headers={"Authorization": f"{cast(str, DAL_GRAPHQL_AUTH_KEY)}"},
+        )
+        self.client = Client(transport=transport, fetch_schema_from_transport=True)
```

Comment:
If we're just issuing queries, I imagine we might not need the schema?

```suggestion
        self.client = Client(transport=transport, fetch_schema_from_transport=False)
```

Reply:
@csande the difference is if we fetch the schema, then the queries are validated before they are sent. From the [docs](https://gql.readthedocs.io/en/stable/usage/validation.html):

_If a GraphQL schema is provided, gql will validate the queries locally before sending them to the backend. If no schema is provided, gql will send the query to the backend without local validation._

There may be some value in this when plugin authors are debugging, since the validation error message may be helpful to them. The downside is that fetching the schema can be slow.

Reply:
Got it, makes sense